user/routes.js

```diff
@@ -0,0 +1,49 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const User = require('./User');
+
+const router = express.Router();
+// so I get from the api and post to the register
+router.get('/', (req, res) => {
+    User.find()
+      .then(users => {
+          res.status(200).json(users);
+      })
+      .catch(err => {
+          res.status(500).json(err);
+      })
+})
+
+router.post('/register', (req, res) => {
+    const newUser = req.body;
+    const { username, password } = req.body;
+    const user = new User(newUser);
+    user.save()
+      .then(user => {
+          res.status(201).json(user);
+      })
+      .catch(err => {
+          res.status(500).json(err);
+      })
+})
+
+router.put('/login', (req, res) => {
+    if (!req.body.username || !req.body.password) {
```

Comment:
If we move this if statement below `const { username, password } = req.body` we can make it more concise:

```js
const { username, password } = req.body

if (!username || !password) return res.status(422).json({ error: 'Please provide a valid username and password.' })
```

Also, don't forget to include `return` so the function execution ends, and the rest of your route's code is ignored.